metabase/metabase
# Add timeline empty states

Epic: https://github.com/metabase/metabase/issues/20289

Changes:
1. Adds an empty state for a case when there are no timelines
2. Adds an empty state for a case when a timeline is empty (no events or all events are archived)
3. Adds an empty state for a case when there are no archived events in a timeline
4. Shows the modal for a timeline instead of showing a list when there is only one timeline

How to test 1:
- Run a new instance or archive all timelines
- Click on the calendar icon in the collection header
- Make sure that the empty state is displayed correctly and it's possible to create a new event with a timeline

<img width="669" alt="Screenshot 2022-02-17 at 16 34 26" src="https://user-images.githubusercontent.com/8542534/154493090-eea5f835-529b-4acb-80ef-62c5a699884d.png">

How to test 2:
- After steps for 1, archive the created event with a menu.
- Make sure that the empty state is displayed correctly
- Make sure there is a menu in the modal header

<img width="672" alt="Screenshot 2022-02-17 at 16 39 15" src="https://user-images.githubusercontent.com/8542534/154493303-f00c9cc9-957d-4fdd-a0e3-efa81e64661f.png">

How to test 3:
- Afters steps for 2, click `View archived events` for this timeline
- `Unarchive` this event.
- Make sure that the empty state is displayed correctly

<img width="683" alt="Screenshot 2022-02-17 at 16 39 23" src="https://user-images.githubusercontent.com/8542534/154493333-22fc4f2b-d249-484d-a0fd-c03fc196948c.png">

How to test 4
- After steps for 4, open `/collection/root/timelines`.
- Make sure that events for the only timeline are shown
- Click `New timeline` in the modal header, fill in all the details and create a new timeline
- Open `/collection/root/timelines` again
- Make sure there is a list of 2 timelines

<img width="663" alt="Screenshot 2022-02-17 at 16 39 54" src="https://user-images.githubusercontent.com/8542534/154493365-84f4f455-23b1-42ff-b938-c78ac039ca56.png">
